Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -341,22 +255,16 @@ - (void)initAndLoadRewardedVideoWithUserId:(NSString *)userId
     }
     
     LogInternal_Internal(@"appId = %@, placementId = %@", appId, placementId);
-    
-    // add to rewarded video delegate map
-    [_rewardedVideoPlacementIdToSmashDelegate setObject:delegate
-                                                 forKey:placementId];
-    
-    // add rewarded video to singleton
-    [[ISVungleAdapterSingleton sharedInstance] addRewardedVideoDelegate:self
-                                                                 forKey:placementId];
-    
+
+    self.rewardedVideoAdapterRouter = [[ISVungleRewardedVideoAdapterRouter alloc] initWithPlacementID:placementId parentAdapter:self delegate:delegate];
+
     switch (_initState) {
         case INIT_STATE_NONE:
         case INIT_STATE_IN_PROGRESS:
             [self initSDKWithAppId:appId];
```

Comment:
Hi @YueVungle dose it make sense to re init the SDK while init is in progress??
i think we should only init SDK if the state is NONE.

i know the IS was doing this and we didnt change it. but we should change it cause SDK will return an error if we try to init while an Init is already in progress.

it should be the same for all add formates.

Reply:
@ManojBudumuru-vungle Hi Manoj, I updated the implementations to below:
```
switch (_initState) {
        case INIT_STATE_NONE:
            [self initSDKWithAppId:appId];
            break;
        case INIT_STATE_IN_PROGRESS:
            [initCallbackDelegates addObject:self];
            break;
```
And in the `initSDKWithAppId:` method,
```
- (void)initSDKWithAppId:(NSString *)appId {
    // add self to the init delegates only in case the initialization has not finished yet
    if (_initState == INIT_STATE_NONE) {
        [initCallbackDelegates addObject:self];
    }
```

We still need `initCallbackDelegates` which is a global static set used for all `ISVungleAdapter` instances, we could think about a scenario that a Rewarded Video is the first `ISVungleAdapter` instance to init the SDK. Then an InitInterstitial ad is requested in the initializing progress, this InitInterstitial ad is another `ISVungleAdapter`  instance，it needs add itself in the `initCallbackDelegates`, because it needs to be notified when completing initialization work. 

PS: From debugging, Aki and I found that one `ISVungleAdapter` instance can hold Interstitial and Banner AdapterRouter at the same time, but no Rewarded Video AdapterRouter.
The  Rewarded Video will be in another `ISVungleAdapter` instance.
Thanks a lot. :) 
cc: @stanley-vungle